src/json/to_json.cr

```diff
@@ -128,6 +128,37 @@ struct Time
   end
 end
 
+struct YAML::Any
+  def to_json(json : JSON::Builder)
+    case object = @raw
+    when Array
+      json.array do
+        each &.to_json(json)
+      end
+    when Hash
+      json.object do
+        each do |key, value|
+          json.field key do
+            value.to_json(json)
+          end
+        end
+      end
+    when "true"
+      json.bool(true)
+    when "false"
+      json.bool(false)
+    when ""
+      json.null
+    when Nil
+      json.null
+    when /^[+-]?([0-9]*[.])?[0-9]+$/
+      object.includes?(".") ? json.number(object.to_f) : json.number(object.to_i)
```

Comment:
Then maybe we shouldn't do this at all, YAML can only emit strings so any YAML being converted JSON can only return JSON of the types that are valid in YAML. The conversion of JSON -> YAML -> JSON simply is lossy by the fact that type wise YAML is a subset of JSON.